src/predevals.js

```diff
@@ -926,6 +917,25 @@ const App = {
 
         return pd
     },
+
+    // utility functions
+
+    /**
+     * Map values of a task id variable to human-readable text.
+     * If this.state.task_id_text is undefined or does not contain the task id variable, returns the input values.
+     *
+     * @param taskId string: the task id variable name
+     * @param taskIdValues array[string]: the task id values to map to text
+     */
+    taskIdValuesToText(taskId, taskIdValues) {
+        const thisState = this.state;
+        if (thisState.task_id_text !== undefined && Object.keys(thisState.task_id_text).includes(taskId)) {
```

Comment:
the check `thisState.task_id_text !== undefined` is new and fixes a bug that showed up if no task id text mapping was provided.  this screenshot verifies that it works in the dev example (horizontal axis has raw location codes):
![image](https://github.com/user-attachments/assets/fad5b717-31af-4db4-9c1a-ce36ead76c74)